ui_testing/testcases/header_tests/test08_usermanagement.py

```diff
@@ -238,4 +227,289 @@ def test010_update_user_role_with_save_cancel_btn(self, save):
+                                               archived_element='user_management:archived')
+
+        self.header_page.select_all_records()
+        self.base_selenium.LOGGER.info('Press on the right menu')
+        self.base_selenium.click(element='user_management:right_menu')
+        self.base_selenium.LOGGER.info('Press on the delete button')
+        self.base_selenium.click(element='user_management:delete')
+        self.header_page.confirm_popup()
+
+        self.base_selenium.LOGGER.info(
+            'message will appear this user related to some data & cant delete it')
+        self.header_page.confirm_popup()
+
+        self.base_selenium.LOGGER.info(
+            'search to make sure this user found in the active table '.format(user_random_name))
+        result = self.header_page.search(value=user_random_name)
+        self.assertTrue(result, user_data)
+
+    def test013_filter_by_user_name(self):
+        """
+        User management Approach: I can filter by user name successfully
+        LIMS-6002
+        :return:
+        """
+        self.base_selenium.click(element='header:user_management_button')
+        random_user_name = self.generate_random_string()
+        random_user_email = self.base_page.generate_random_email()
+        self.header_page.create_new_user(user_name=random_user_name, user_email=random_user_email,
+                                         user_role='', user_password='1', user_confirm_password='1')
+
+        self.base_selenium.click(element='general:menu_filter_view')
+        user_filter = self.header_page.filter_user_by(filter_element='user_management:filter_name',
+                                                      filter_text=random_user_name)
+        result_user = self.header_page.result_table()[0]
+        self.assertTrue(result_user, user_filter)
+        self.base_selenium.click(element='user_management:filter_reset_btn')
+
+    def test014_filter_by_email(self):
+        """
+        User management Approach: I can filter by user email successfully
+        LIMS-6442
+        :return:
+        """
+        self.base_selenium.click(element='header:user_management_button')
+        random_user_name = self.generate_random_string()
+        random_email = self.header_page.generate_random_email()
+        self.header_page.create_new_user(user_name=random_user_name, user_email=random_email,
+                                         user_role='', user_password='1', user_confirm_password='1')
+
+        self.base_selenium.click(element='general:menu_filter_view')
+        user_filter = self.header_page.filter_user_by(filter_element='user_management:filter_email',
+                                                      filter_text=random_email)
+
+        result_user = self.header_page.result_table()[0]
+        self.assertTrue(result_user, user_filter)
+        self.base_selenium.click(element='user_management:filter_reset_btn')
```

Comment:
parametrized with the previous one

Reply:
@0xIslamTaha I need each filter test case separate same as we agreed